One remaining issue - linting error:

```
pkg/discovery/network/weavenet.go:26:1: directive `// nolint:nilnil // Intentional as the purpose is to discover.` is unused for linter "nilnil" (nolintlint)
```

The code no longer returns `nil, nil` so the linting exception needs to be removed.

--- a/pkg/discovery/network/weavenet.go
+++ b/pkg/discovery/network/weavenet.go
@@ -23,7 +23,6 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
-// nolint:nilnil // Intentional as the purpose is to discover.
 func discoverWeaveNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, error) {
 	weaveNetPod, err := FindPod(clientSet, "name=weave-net")
 
@@ -31,23 +30,17 @@ func discoverWeaveNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, erro
 		return nil, err
 	}
 
-	var clusterNetwork *ClusterNetwork
+	clusterNetwork := &ClusterNetwork{
+		NetworkPlugin: cni.WeaveNet,
+	}
 
 	for i := range weaveNetPod.Spec.Containers {
 		for _, envVar := range weaveNetPod.Spec.Containers[i].Env {
 			if envVar.Name == "IPALLOC_RANGE" {
-				clusterNetwork = &ClusterNetwork{
-					PodCIDRs:      []string{envVar.Value},
-					NetworkPlugin: cni.WeaveNet,
-				}
-
+				clusterNetwork.PodCIDRs = []string{envVar.Value}
 				break
 			}
 		}
-	}
-
-	if clusterNetwork == nil {
-		return nil, nil
 	}
 
 	clusterIPRange, err := findClusterIPRange(clientSet)

